[Detection Engine][Docs][8.18] New wording for frozen tier support (#6659)
elastic/security-docs: docs/detections/detection-engine-intro.asciidoc

[float]
[[cold-tier-detections]]
== Manage data in cold and frozen tiers

Cold data tiers store time series data that is accessed infrequently and rarely updated, while frozen data tiers hold time series data that is accessed even less frequently and never updated. If you are automating searches across different {ref}/data-tiers.html[data tiers] using rules, consider the following best practices and limitations.

[float]
[[best-practices-data-tiers]]
=== Best practices 

> * **Retention in hot tier**: We recommend keeping data in the hot tier for at least 24 hours. {ilm-cap} policies that roll over data more frequently than once every 24 hours can increase the volume of frozen data queried by rules, leading to performance issues.

Reviewer: @MikePaquette @approksiu want to get your signoff here that what we're saying we support is 24 hours in hot, no less.

Reviewer 2: Recommendation makes sense. Do you want to use "support" in the wording here?

Reviewer 2: @nastasha-solomon Mike's suggestion: "We recommend keeping ingested data in the hot data tier (ILM hot phase) for at least 24 hours. ILM policies that move ingested data from the hot phase to another phase (e.g., cold, frozen) sooner than 24 hours may cause performance issues and/or rule execution errors."

Author: Thanks, @approksiu! 

```suggestion
* **Retention in hot tier**: We recommend keeping data in the hot data tier ({ilm-cap} hot phase) for at least 24 hours. {ilm-cap} policies that move ingested data from the hot phase to another phase (for example, cold or frozen) sooner than 24 hours may cause performance issues and/or rule execution errors.
```

cc: @marshallmain @yctercero since there's a related discussion on this description below.

Reviewer: I like Mike's suggestion! 👍🏽

Reviewer 3: grammatically I would just swap `sooner than 24 hours` -> `in less than 24 hours`, otherwise it looks good!

Author: Agree @marshallmain - thanks, all!

```suggestion
* **Retention in hot tier**: We recommend keeping data in the hot data tier ({ilm-cap} hot phase) for at least 24 hours. {ilm-cap} policies that move ingested data from the hot phase to another phase (for example, cold or frozen) in less than 24 hours may cause performance issues and/or rule execution errors.
```

> * **Retention in hot tier**: We recommend keeping data in the hot tier for at least 24 hours. {ilm-cap} policies that roll over data more frequently than once every 24 hours can increase the volume of frozen data queried by rules, leading to performance issues.

Reviewer: ```suggestion
* **Retention in hot tier**: We recommend keeping data in the hot tier for at least 24 hours. {ilm-cap} policies that roll over data more frequently than once every 24 hours can increase the volume of frozen data queried by rules, missing out on the performance enhancements offered by hot tier.
```

cc @marshallmain thoughts on the change here? Was trying to make the wording clear that the performance issues rest on the ES side with how frozen works as opposed to something on our end we're doing with frozen data that results in performance issues.

Reviewer 3: Perhaps something like `leading to slower queries and potentially overloading (overwhelming? resource exhaustion?) the frozen tier`? I think calling hot tier a performance enhancement sounds a bit like it's a new feature which could be confusing.

Reviewer: Yea, I like the suggestion.

Author: Resolving since we discussed this above.

> * **Replicas for mission-critical data**: Your data should have replicas if it must be constantly available. Since frozen tiers don't support replicas, shard unavailability can cause partial rule run failures. Shard unavailability may be also encountered during or after {stack} upgrades. If this happens, you can <<manually-run-rules,manually run>> the rule over the affected time period.

Reviewer 3: ```suggestion
* **Replicas for mission-critical data**: Your data should have replicas if it must be highly available. Since frozen tiers don't support replicas, shard unavailability can cause partial rule run failures. Shard unavailability may be also encountered during or after {stack} upgrades. If this happens, you can <<manually-run-rules,manually rerun>> rules over the affected time period once the shards are available.
```
"highly available" sounds a bit odd but is the [standard term](https://en.wikipedia.org/wiki/High_availability)

Author: Looks to also be used in Elasticsearch docs about cluster availability: https://www.elastic.co/guide/en/elasticsearch/reference/current/high-availability.html

> * **Replicas for Mission-Critical Data**: Your data should have replicas if it must be constantly available. Since frozen tiers don't support replicas, shard unavailability can cause partial rule run failures. Shard unavailability may be also encountered during or after {stack} upgrades. If this happens, you can re-run the rule over the affected time period using <<manually-run-rules,manual runs>>.

Author: ```suggestion
* **Replicas for Mission-Critical Data**: Your data should have replicas if it must be constantly available. Since frozen tiers don't support replicas, shard unavailability can cause partial rule run failures. Shard unavailability may be also encountered during or after {stack} upgrades. If this happens, you can <<manually-run-rules,manually run>> the rule over the affected time period.
```

> * **Replicas for mission-critical data**: Your data should have replicas if it must be highly available. Since frozen tiers don't support replicas, shard unavailability can cause partial rule run failures. Shard unavailability may be also encountered during or after {stack} upgrades. If this happens, you can <<manually-run-rules,manually rerun>> rules over the affected time period once the shards are available.

Reviewer 4: ```suggestion
* **Replicas for mission-critical data**: Your data should have replicas if it must be highly available. Since frozen tiers don't support replicas, shard unavailability can cause partial rule run failures if they query frozen tiers. Shard unavailability may also occur during or after {stack} upgrades. If this happens, you can <<manually-run-rules,manually rerun>> rules over the affected time period once the shards are available.
```

Author: Some additional tweaks for clarity:

```suggestion
* **Replicas for mission-critical data**: Your data should have replicas if it must be highly available. Since frozen tiers don't support replicas, shard unavailability can cause partial rule run failures if rules query frozen tiers. Shard unavailability may also occur during or after {stack} upgrades. If this happens, you can <<manually-run-rules,manually rerun>> rules over the affected time period once the shards are available.

```

[float]
[[limitations-data-tiers]]
=== Limitations

Data tiers are a powerful and useful tool. When using them, keep the following limitations in mind:

> * {ilm-cap} policies for indices controlled by {elastic-sec}, including alerts and list indices, must not be modified.

Reviewer 4: ```suggestion
* You cannot modify {ilm-cap} policies for indices controlled by {elastic-sec}, including alerts and list indices.
```
This suggestion could be wrong, but as written before it sounded like you _could_ but shouldn't modify them. I'm not sure which is true, but if it's a "can but shouldn't" situation, I wonder why I shouldn't! Maybe we could explain it in brief?

Reviewer: It's a "can but shouldn't" - not sure this would be a good spot to get into all the "whys" as it could take away from just focusing on the tiers.

Author: What if we go in the direction of encouraging users to work with the default ILM settings? For example:

```suggestion
* {ilm-cap} policies for {elastic-sec}-controlled indices, such as alert and list indices, function best with their default settings and should not be modified.
```

Reviewer 3: We've gotten many SDHs where rules were broken due to modifications to ILM policies so I'd prefer stronger language, e.g. "modifying ILM policies for elastic controlled indices may cause rule failures"

Author: How's this, @marshallmain?

```suggestion
* To avoid rule failures, do not modify {ilm-cap} policies for {elastic-sec}-controlled indices, such as alert and list indices.
```

Reviewer 3: I like that!

> * Indicator match rule performance can be severely impacted by querying data in frozen tiers.

Reviewer 4: From what it says above about replicas for mission critical data, it sounds like this might apply to any rule type. Is that wrong?

Reviewer: Are you asking about performance implications? If so, yes, there are performance implications any time. you query frozen, but with Indicator Match, it is especially pronounced. 

Ideally, users won't need to think about what tier they are querying, but for now we do have to be aware of it.

Author: @yctercero can we actually removed this?

> * Indicator match rule performance can be severely impacted by querying data in frozen tiers.

In the ["Limited support for indicator match rules"](https://security-docs_bk_6659.docs-preview.app.elstc.co/guide/en/security/8.x/detection-engine-overview.html#support-indicator-rules) section, we make a similar statement:

> Indicator match rules don’t support cold or frozen data. Cold or frozen data in indices queried by indicator match rules must be older than the time range queried by the rule. If your data’s timestamps are unreliable, you can exclude cold and frozen tier data using a [Query DSL filter](https://security-docs_bk_6659.docs-preview.app.elstc.co/guide/en/security/8.x/exclude-cold-frozen-data-individual-rules.html).

Reviewer: I'd be ok with that! 

cc @marshallmain @approksiu

Reviewer 3: Yeah I think we can remove it

Author: ```suggestion
```

> * Cold and frozen source data must have an {ilm} policy that keeps it in the hot or warm tiers for at least one day.

Reviewer 3: ```suggestion
* Source data must have an {ilm} policy that keeps it in the hot or warm tiers for at least one day before moving to cold or frozen tiers.
```

[float]
[[best-practices-data-tiers]]
=== Best practices 

* **Retention in hot tier**: We recommend keeping data in the hot data tier ({ilm-cap} hot phase) for at least 24 hours. {ilm-cap} policies that move ingested data from the hot phase to another phase (for example, cold or frozen) in less than 24 hours may cause performance issues and/or rule execution errors.
* **Replicas for mission-critical data**: Your data should have replicas if it must be highly available. Since frozen tiers don't support replicas, shard unavailability can cause partial rule run failures if rules query frozen tiers. Shard unavailability may also occur during or after {stack} upgrades. If this happens, you can <<manually-run-rules,manually rerun>> rules over the affected time period once the shards are available.

[float]
[[limitations-data-tiers]]
=== Limitations

Data tiers are a powerful and useful tool. When using them, keep the following limitations in mind:

* {ilm-cap} policies for indices controlled by {elastic-sec}, including alerts and list indices, must not be modified.
* Indicator match rule performance can be severely impacted by querying data in frozen tiers.

> * Source data must have an {ilm} policy that keeps it in the hot or warm tiers for at least one day before moving to cold or frozen tiers.

Author: To keep the lang consistent.

```suggestion
* Source data must have an {ilm} policy that keeps it in the hot or warm tiers for at least 24 hours before moving to cold or frozen tiers.
```
